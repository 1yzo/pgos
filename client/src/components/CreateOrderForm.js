import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import 'react-day-picker/lib/style.css';
import ReactModal from 'react-modal';
import '../styles/order-form.css';
import { startSetPageCount } from '../actions/pageCount';
import { startSetPage } from '../actions/currentPage';

export class CreateOrderForm extends React.Component {
    state = {
        name: '',
        brewMethod: '',
        shipDate: moment().toDate(),
        cases: '',
        packetsPerCase: 25,
        notes: '',
        priority: false,
    }

    closeModal = () => {
        this.props.closeModal();
    }
    
    handleSubmit = (e) => {
        e.preventDefault();
        const order = this.state;
        if (this.state.name && this.state.brewMethod && this.state.shipDate && this.state.cases && this.state.packetsPerCase) {
            fetch('/orders/', {
                method: 'POST',
                body: JSON.stringify({ ...order, shipDate: moment(this.state.shipDate).valueOf() }),
                headers: { 'content-type': 'application/json' }
            })
                .then(() => {
                    this.props.dispatch(startSetPageCount());
                })
                .then(() => {
                    if (this.props.currentPageIndex !== this.props.pageCount) {
                        this.props.dispatch(startSetPage(this.props.currentPageIndex))
                            .then(() => this.props.closeModal());
                    }
                });
        } 
    }

    handleChange = (e) => {
        const field = e.target.id;
        const value = e.target.value;
        const nextState = { ...this.state };
        nextState[field] = value;
        this.setState(() => nextState);
    }
    
    handlePriorityChange = () => {
        this.setState((prevState) => ({ priority: !prevState.priority }));
    }
    
    render() {
        return (
            <ReactModal
                className="order-form"
                isOpen={this.props.showModal}
                contentLabel="Modal"
                closeTimeoutMS={300}
                shouldCLoseOnOverlayClick={true}
                shouldCloseOnEsc={true}
                ariaHideApp={false}
            >
                <form className="order-form__form"onSubmit={this.handleSubmit}>
                    <div className="exit" onClick={this.closeModal}>X</div>
                    <div className="order-form__info">Perfectly Ground Work Orders</div>
                    <div className="order-form__sub-info">
                        Instructional Text would go here - Lorem Ipsum dolor sit amet, consectetur ad
                        Ipsum dolor sit amet, consectetur ad
                        </div>
                    <div className="order-form__container">
                        <div className="order-form__section">
                            <label>Coffee</label>
                            <input id="name" type="text" value={this.state.name} onChange={this.handleChange} />
                        </div>
                        <div className="order-form__section">
                            <label>Brew Method</label>
                            <select id="brewMethod" onChange={this.handleChange}>
                                <option value="">Choose One</option>
                                <option value="Aeropress">Aeropress</option>
                                <option value="Coffee Maker">Coffee Maker</option>
                                <option value="Cold Brew">Cold Brew</option>
                                <option value="French Press">French Press</option>
                                <option value="Pour Over">Pour Over</option>
                            </select>
                        </div>
                    </div>
                    <div className="order-form__container">
                        <div className="order-form__section">
                            <label>Ship Date</label>
                            <DayPickerInput
                                value={this.state.shipDate} 
                                onDayChange={(shipDate) => this.setState(() => ({ shipDate }))}
                            />
                        </div>
                        <div className="order-form__section">
                            <label>Number of Cases</label>  
                            <input
                                id="cases"
                                type="text"
                                value={this.state.cases}
                                onChange={this.handleChange}
                                placeholder='#'
                            />
                        </div>
                        <div className="order-form__section">
                            <label>Packets per Case</label>
                            <select id="packetsPerCase" onChange={this.handleChange}>
                                <option value={25}>25</option>
                                <option value={50}>50</option>
                            </select>
                        </div>
                    </div>
                    <label>Notes</label>
                    <textarea id="notes" value={this.state.notes} onChange={this.handleChange} />
                    <div style={{ marginBottom: '2rem' }}>
                        <input id="priority" type="checkbox" onChange={this.handlePriorityChange} />
                        <label htmlFor="priority">Priority</label>
                    </div>
                    <button>Submit Work Order</button>
                </form>
            </ReactModal>
        )
    }
}

const mapStateToProps = (state) => ({
    currentPageIndex: Number(state.currentPage.index),
    pageCount: state.pageCount
});

const mapDispatchToProps = (dispatch) => ({
    submitOrder: (order) => {
        
    }
});

export default connect(mapStateToProps)(CreateOrderForm);