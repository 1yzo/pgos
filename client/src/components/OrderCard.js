import React from 'react';
import moment from 'moment';

const OrderCard = (props) => (
    <div className="order-card" style={{background: props.index % 2 === 0 ? '#fff' : '#eee'  }}>
        <div>{props.order.name}</div>
        <div>{props.order.brewMethod}</div>
        <div>{props.order.cases}</div>
        <div>{props.order.packetsPerCase}</div>
        <div className="order-card__date">
            <div>{moment(props.order.shipDate).format('MM/DD/YYYY')}</div>
            {props.order.priority && <div className="order-card__star">*</div>}
        </div>
        <div className="order-card__number">#57120</div>
        <i className="material-icons" onClick={() => props.selectOrder(props.order)}>remove_red_eye</i>
    </div>
);

export default OrderCard;