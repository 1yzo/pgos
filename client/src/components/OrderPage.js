import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { startSetPage } from '../actions/currentPage';

class OrderPage extends React.Component {    


    componentWillReceiveProps(nextProps) {
        const { pageIndex } = nextProps.match.params;
        if (pageIndex !== this.props.match.params.pageIndex)
        this.props.dispatch(startSetPage(pageIndex)); 
    }
    
    render() {
        const { orders } = this.props;
        return (
            <div className="orders-container">
                {orders.map((order) => (
                    <div key={order._id}>
                        <div>{order.name}</div>
                    </div>
                ))}
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    orders: state.currentPage.orders
});

export default withRouter(connect(mapStateToProps)(OrderPage));