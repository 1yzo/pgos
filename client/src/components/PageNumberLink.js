import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const PageNumberLink = ({ index, pageCount, currentPageIndex }) => {
    if (index < pageCount) {
        return (
            <div >
                <Link className={"page-link" + (Number(currentPageIndex) === index ? ' page-link--active' : '')}to={`/orders/${index}`}>{index}</Link>
            </div>
        );
    } else return null;
};

const mapStateToProps = (state) => ({
    pageCount: state.pageCount,
    currentPageIndex: state.currentPage.index
});

export default connect(mapStateToProps)(PageNumberLink);