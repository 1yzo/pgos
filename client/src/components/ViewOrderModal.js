import React from 'react';
import ReactModal from 'react-modal';
import moment from 'moment';
import '../styles/order-view.css';

class ViewOrderModal extends React.Component {
    render() {
        const { order } = this.props; 
        return (
            <ReactModal
                className="order-view"
                isOpen={this.props.showModal}
            >
                <div className="exit" onClick={() => this.props.closeModal()}>X</div>
                <div className="order-view__section">
                    <div className="order-view__label">Coffee</div>
                    <div className="order-view__info">{order.name}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Brew Method</div>
                    <div className="order-view__info">{order.brewMethod}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Ship Date</div>
                    <div className="order-view__info">{moment(order.shipDate).format('MM/DD/YYYY')}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Number of Cases</div>
                    <div className="order-view__info">{order.cases}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Packets per Case</div>
                    <div className="order-view__info">{order.packetsPerCase}</div>
                </div>
                {order.notes &&
                <div className="order-view__section">
                    <div className="order-view__label order-view__label--notes">Notes</div>
                    <div className="order-view__info">{order.notes}</div>
                </div>}
            </ReactModal>
        );
    }
}

export default ViewOrderModal;