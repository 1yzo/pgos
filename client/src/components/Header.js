import React from 'react';
import moment from 'moment';
import '../styles/header.css';
import CreateOrderForm from './CreateOrderForm';

class Header extends React.Component {
    state = { showModal: false };
    
    openModal = () => {
        this.setState(() => ({ showModal: true }));
    };

    closeModal = () => {
        this.setState(() => ({ showModal: false }));
    };

    render() {
        return (
            <div className="header">
                <div className="header__title">BLUE BOTTLE COFFEE</div>
                <div className="header__container">
                    <div className="header__date">
                        <div className="header__date__month">{moment().format('MMM').toUpperCase()}</div>
                        <div className="header__date__day">{moment().format('DD')}</div>
                    </div>
                    <div className="header__subtitle">Perfectly Ground Work Orders</div>
                    <button className="header__button" onClick={this.openModal}>CREATE ORDER</button>
                    </div>
                <CreateOrderForm showModal={this.state.showModal} closeModal={this.closeModal}/>
            </div>
        );
    }
}

export default Header;