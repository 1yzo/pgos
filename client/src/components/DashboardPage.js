import React from 'react';
import { connect } from 'react-redux';
import { Route, Link } from 'react-router-dom'; 
import OrderPage from './OrderPage';
import Header from './Header'
import PageNumberLink  from './PageNumberLink';
import '../styles/dashboard-page.css';

export class DashboardPage extends React.Component {
    componentDidMount() {
        this.props.history.push('/orders/0');
    }
    
    render() {
        const { currentPageIndex, pageCount } = this.props;

        const initialIndex = currentPageIndex - (currentPageIndex % 5);
        return (
            <div className="dashboard-page">
                <Header />
                <Route path="/orders/:pageIndex" component={OrderPage} />
                <div style={{ display: 'flex', justifyContent: 'center' }}>
                    <div className="controls">
                        <Link className="page-link" to={`/orders/${0}`}>{'<<'}</Link>
                        <div onClick={this.handlePrevClick}>
                            {currentPageIndex >= 1 &&<Link className="page-link"to={`/orders/${Number(currentPageIndex) - 1}`}>Prev</Link>}
                        </div>
                        <PageNumberLink index={initialIndex} />
                        <PageNumberLink index={initialIndex + 1} />
                        <PageNumberLink index={initialIndex + 2} />
                        <PageNumberLink index={initialIndex + 3} />
                        <PageNumberLink index={initialIndex + 4} />
                        {currentPageIndex < pageCount - 1 && <Link className="page-link" to={`/orders/${Number(currentPageIndex) + 1}`}>Next</Link>}
                        <Link className="page-link" to={`/orders/${pageCount - 1}`}>{`>>`}</Link>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    currentPageIndex: Number(state.currentPage.index),
    pageCount: state.pageCount
});

export default connect(mapStateToProps)(DashboardPage);