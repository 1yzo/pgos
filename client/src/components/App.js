import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import CreateOrderForm from './CreateOrderForm';
import DashboardPage from './DashboardPage';

class App extends Component {
	render() {
		return (
			<BrowserRouter>
				<Switch>
					<Route path="/" component={DashboardPage} />
				</Switch>
      		</BrowserRouter>
    	);
 	}
}

export default App;