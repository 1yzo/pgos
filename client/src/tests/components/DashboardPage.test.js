import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPage } from '../../components/DashboardPage';

let history, wrapper;

beforeEach(() => {
    history = { push: jest.fn() }
    wrapper = shallow(<DashboardPage history={history} />)
});

test('should render DashboardPage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});