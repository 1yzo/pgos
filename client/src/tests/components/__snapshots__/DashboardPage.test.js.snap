// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashboardPage
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Route
          component={[Function]}
          path="/orders/:pageIndex"
/>,
        <div
          style={
                    Object {
                              "display": "flex",
                              "justifyContent": "center",
                            }
          }
>
          <div
                    className="controls"
          >
                    <Link
                              className="page-link"
                              replace={false}
                              to="/orders/0"
                    >
                              &lt;&lt;
                    </Link>
                    <div
                              onClick={undefined}
                    />
                    <Connect(Component)
                              index={NaN}
                    />
                    <Connect(Component)
                              index={NaN}
                    />
                    <Connect(Component)
                              index={NaN}
                    />
                    <Connect(Component)
                              index={NaN}
                    />
                    <Connect(Component)
                              index={NaN}
                    />
                    <Link
                              className="page-link"
                              replace={false}
                              to="/orders/NaN"
                    >
                              &gt;&gt;
                    </Link>
          </div>
</div>,
      ],
      "className": "dashboard-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/orders/:pageIndex",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="controls"
>
            <Link
                        className="page-link"
                        replace={false}
                        to="/orders/0"
            >
                        &lt;&lt;
            </Link>
            <div
                        onClick={undefined}
            />
            <Connect(Component)
                        index={NaN}
            />
            <Connect(Component)
                        index={NaN}
            />
            <Connect(Component)
                        index={NaN}
            />
            <Connect(Component)
                        index={NaN}
            />
            <Connect(Component)
                        index={NaN}
            />
            <Link
                        className="page-link"
                        replace={false}
                        to="/orders/NaN"
            >
                        &gt;&gt;
            </Link>
</div>,
          "style": Object {
            "display": "flex",
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="page-link"
                replace={false}
                to="/orders/0"
>
                &lt;&lt;
</Link>,
              <div
                onClick={undefined}
/>,
              <Connect(Component)
                index={NaN}
/>,
              <Connect(Component)
                index={NaN}
/>,
              <Connect(Component)
                index={NaN}
/>,
              <Connect(Component)
                index={NaN}
/>,
              <Connect(Component)
                index={NaN}
/>,
              false,
              <Link
                className="page-link"
                replace={false}
                to="/orders/NaN"
>
                &gt;&gt;
</Link>,
            ],
            "className": "controls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "<<",
                "className": "page-link",
                "replace": false,
                "to": "/orders/0",
              },
              "ref": null,
              "rendered": "<<",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": false,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": false,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": NaN,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": NaN,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": NaN,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": NaN,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "index": NaN,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": ">>",
                "className": "page-link",
                "replace": false,
                "to": "/orders/NaN",
              },
              "ref": null,
              "rendered": ">>",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Route
            component={[Function]}
            path="/orders/:pageIndex"
/>,
          <div
            style={
                        Object {
                                    "display": "flex",
                                    "justifyContent": "center",
                                  }
            }
>
            <div
                        className="controls"
            >
                        <Link
                                    className="page-link"
                                    replace={false}
                                    to="/orders/0"
                        >
                                    &lt;&lt;
                        </Link>
                        <div
                                    onClick={undefined}
                        />
                        <Connect(Component)
                                    index={NaN}
                        />
                        <Connect(Component)
                                    index={NaN}
                        />
                        <Connect(Component)
                                    index={NaN}
                        />
                        <Connect(Component)
                                    index={NaN}
                        />
                        <Connect(Component)
                                    index={NaN}
                        />
                        <Link
                                    className="page-link"
                                    replace={false}
                                    to="/orders/NaN"
                        >
                                    &gt;&gt;
                        </Link>
            </div>
</div>,
        ],
        "className": "dashboard-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/orders/:pageIndex",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="controls"
>
              <Link
                            className="page-link"
                            replace={false}
                            to="/orders/0"
              >
                            &lt;&lt;
              </Link>
              <div
                            onClick={undefined}
              />
              <Connect(Component)
                            index={NaN}
              />
              <Connect(Component)
                            index={NaN}
              />
              <Connect(Component)
                            index={NaN}
              />
              <Connect(Component)
                            index={NaN}
              />
              <Connect(Component)
                            index={NaN}
              />
              <Link
                            className="page-link"
                            replace={false}
                            to="/orders/NaN"
              >
                            &gt;&gt;
              </Link>
</div>,
            "style": Object {
              "display": "flex",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="page-link"
                  replace={false}
                  to="/orders/0"
>
                  &lt;&lt;
</Link>,
                <div
                  onClick={undefined}
/>,
                <Connect(Component)
                  index={NaN}
/>,
                <Connect(Component)
                  index={NaN}
/>,
                <Connect(Component)
                  index={NaN}
/>,
                <Connect(Component)
                  index={NaN}
/>,
                <Connect(Component)
                  index={NaN}
/>,
                false,
                <Link
                  className="page-link"
                  replace={false}
                  to="/orders/NaN"
>
                  &gt;&gt;
</Link>,
              ],
              "className": "controls",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "<<",
                  "className": "page-link",
                  "replace": false,
                  "to": "/orders/0",
                },
                "ref": null,
                "rendered": "<<",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": false,
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": false,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "index": NaN,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "index": NaN,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "index": NaN,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "index": NaN,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "index": NaN,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": ">>",
                  "className": "page-link",
                  "replace": false,
                  "to": "/orders/NaN",
                },
                "ref": null,
                "rendered": ">>",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
